core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -489,6 +499,36 @@ public HTTPClient build() {
       withHeader(CLIENT_VERSION_HEADER, IcebergBuild.fullVersion());
       withHeader(CLIENT_GIT_COMMIT_SHORT_HEADER, IcebergBuild.gitCommitShortId());
 
+      String proxyHostname =
+          PropertyUtil.propertyAsString(properties, HTTPClient.REST_PROXY_HOSTNAME, null);
+
+      Integer proxyPort =
+          PropertyUtil.propertyAsNullableInt(properties, HTTPClient.REST_PROXY_PORT);
+
+      if (proxyHostname != null && !proxyHostname.isEmpty() && proxyPort != null) {
```

Comment:
you can use `!Strings.isNullOrEmpty(proxyHostname)`. Same further below with username/password